### ISSUE TITLE
Test against Blob Only storage accounts for blob trigger/input/output

### ISSUE THREAD

**Issue by christopheranderson** [Azure/azure-webjobs-sdk#810]
We need to decide if we support blob only storage accounts for blob trigger/input/output.

**Comment by mathewc** [comment]
I think this is a duplicate of https://github.com/Azure/azure-webjobs-sdk/issues/797.

**Comment by lindydonna** [comment]
@mathewc I think you're right. This issue was to track whether or not we wanted to support the scenario at all, especially because there's probably a perf impact. Are you saying that we should decide that we are disallowing them and update the portal pickers appropriately?  We do have issues to track the portal work FYI.

**Comment by mathewc** [comment]
Blob only storage accounts can't be used for our "primary" storage account, since we use queues internally. They should be supported in general though, however there have been some core SDK bugs logged for issues recently we may need to investigate there.

**Comment by lindydonna** [comment]
In that case, this is not a duplicate. The other issue is about showing an error in the host (if I understood it correctly), whereas this is about determine whether we care to support the scenario. (I.e., let's firm up the "should work" statement). The easiest option would be to just decide not to support for RC and revisit later.

**Comment by lindydonna** [comment]
This scenario probably works, but we don't want to support it for RC. There is a related issue for filtering out blob-only accounts from the portal picker.

**Comment by paulbatum** [comment]
@mamaso Confirm blob only works at the SDK level. If it does, then reactivate https://github.com/Azure/azure-webjobs-sdk-script/issues/773

**Comment by mamaso** [comment]
Investigated, blob-only storage accounts work without a hitch.

The premium storage account issues are due to a bug with GetServicePropertiesAsync in the storage sdk: https://github.com/Azure/azure-storage-net/issues/354

We use GetServicePropertiesAsync to do account validation & enable blob logging.  
Workarounds: I could handle this particular exception, assuming that it's caused by premium storage, or we can wait for storage to fix the bug.

Also put in a request for an api to determine storage account type (general, blob only, premium)
https://github.com/Azure/azure-storage-net/issues/355

**Comment by lindydonna** [comment]
@davidebbo and @paulbatum: The Storage team closed both of the SDK issues that @mamaso opened, since the Storage REST API doesn't actually support these features.

My 2 cents: catching the exception on GetServicePropertiesAsync and assuming it's a premium account seems hacky and fragile. I vote to keep the exception in place. 

Unfortunately, it looks like we can't filter these accounts from the portal either, due to Azure/azure-storage-net#355.

**Comment by davidebbo** [comment]
To me, it feels unfortunate to make it impossible for users to use those premium accounts when we have demonstrated that it can work great, and is only getting blocked by what looks like a technicality.

**Comment by paulbatum** [comment]
I'm with David, we should enable the scenario, even if it requires a hacky workaround.

**Comment by mamaso** [comment]
Alright, I'll hack it in 👍 

Only potential issue I see is that premium accounts won't work with blob logging & may have problems with blob container scans as well, @brettsam where do we store the scan records? all in the primary account or in the account being scanned?

**Comment by lindydonna** [comment]
@mamaso I reopened https://github.com/Azure/azure-webjobs-sdk-script/issues/773. If it turns out nothing needs to change in the script host, then please close that issue once the fix is deployed and the original customer can test it.

**Comment by paulbatum** [comment]
@mamaso If thats the case, it sounds like premium storage will work for input/output but not for trigger. Make sure we handle each case correctly!

**Comment by brettsam** [comment]
The scan records would all be in the primary account (that would include the blobreceipts and the blobscaninfo). And the queue that we ultimately write to should also be in the primary account.

**Comment by mamaso** [comment]
Thanks @brettsam, the blob scanner should work.  However, log scanning will not.  @paulbatum @mathewc Are we ok with enabling premium blob triggers if they only use blob scanning?

**Comment by paulbatum** [comment]
Isn't the scale controller currently dependent on log scanning? To provide trigger support it needs to work in dynamic.

**Comment by mamaso** [comment]
@paulbatum yep you're right, from the webjobs perspective it's not a problem but with functions it definitely is.  I'll be conservative and disable these accounts as blob triggers.

**Comment by lindydonna** [comment]
Fix is merged
